pkg/istio-agent/sds-agent.go

```diff
@@ -293,13 +297,19 @@ func (sa *SDSAgent) Start(isSidecar bool, podNamespace string) (*sds.Server, err
 	serverOptions.PilotCertProvider = sa.PilotCertProvider
 	// Next to the envoy config, writeable dir (mounted as mem)
 	serverOptions.WorkloadUDSPath = LocalSDS
-	serverOptions.UseLocalJWT = sa.CertsPath == "" // true if we don't have a key.pem
 	serverOptions.CertsDir = sa.CertsPath
 	serverOptions.JWTPath = sa.JWTPath
 	serverOptions.OutputKeyCertToDir = sa.OutputKeyCertToDir
 	serverOptions.CAEndpoint = sa.CAEndpoint
 	serverOptions.TLSEnabled = sa.RequireCerts
 	serverOptions.ClusterID = sa.ClusterID
+	serverOptions.FileMountedCerts = sa.FileMountedCerts
+	// If proxy is using file mounted certs, JWT token is not needed.
+	if sa.FileMountedCerts {
+		serverOptions.UseLocalJWT = false
```

Comment:
According to security/pkg/nodeagent/sds/server.go: 
// UseLocalJWT is set when the sds server should use its own local JWT, and not expect one
// from the UDS caller.

Based on the above comment, when UseLocalJWT is false, SDS server will not use its local JWT (but may still expect JWT from the UDS caller). Can you test that setting "serverOptions.UseLocalJWT = false" will skip reading JWT token?